daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltin.scala

```diff
@@ -1397,8 +1397,31 @@ private[lf] object SBuiltin {
         case SOptional(opt) =>
           opt match {
             case None =>
+              machine.ledgerMode match {
+                case OffLedger => ()
```

Comment:
This seems off, we don’t expect to ever hit this do we? I think this should extend `OnLedgerBuiltin` and crash in off-ledger mode.

Reply:
Yes, I think you are right.
For some reason I was thinking the exception handling could happen on or off ledger.
But I support the fact that `try`..`catch` is typed as an `Update` prevent this.
I'll change to crash as you suggest.

Reply:
It will happen off ledger but it won’t happen via the speedy builtin.

Reply:
What. You can have _off_ ledger execution of `Update` expressions?
I'm confused now!

Reply:
No you can’t have off-ledger execution of `Update`. But you can throw an exception outside of `Update` and we will add a `Script`-specific catch (which won’t go via a speedy builtin, see https://github.com/digital-asset/daml/pull/7368 for an ancient prototype).

Reply:
Ok. that's clear.  A throw can happen anywhere. But the `try-catch` (`Update` expression) can only be on ledger.

And it is the speedy execution of the `try-catch` which performs the `{begin,end,abort,rollback}Try`, which is where the `OnLedger` is required, and so may be assumed.

Reply:
done!